### ISSUE TITLE
styled forwardRef argument works, but there are no types for it

### ISSUE THREAD

**Issue by matthewlucock** [cristianbote/goober#176]
Hi @cristianbote! Thanks for this project! This is probably just a simple oversight, but I've noticed that despite `styled` taking a second argument `forwardRef`, and it being documented as such, there are no types for that argument; [every version of the function](https://github.com/cristianbote/goober/blob/a630943ca6ccbb422607ee9dc0c28bec14be9429/goober.d.ts#L8) is typed as only taking a single argument. If I create a component using `styled as any`, the `forwardRef` argument behaves at runtime exactly as I expect.

**Comment by cristianbote** [comment]
Hey @matthewlucock, thank you so much for your kind words and for opening this issue. Much appreciate it!

Indeed 😬  it's something I overlooked. Would you like to open up a PR for it? If not I believe I can do it later on. Feel free to ping me on slack if you need more info.

Have a great one and thank you!

**Comment by cristianbote** [comment]
@devrnt do you think you could tackle this one as well? Please do not feel as something you have to, I only pinged you as you recently added those sweet types for the `DefaultTheme`

**Comment by devrnt** [comment]
Seems an easy fix, is it okay to make this `forwardRef` of type `any`?

**Comment by michalczaplinski** [comment]
Mind if I send a PR? :) I've already done a quick fix with this. 

As we support not only react but also other view libraries, I don't think we can make the type more specific than this:

```
type ForwardRefFunction = {
        (props: any, ref: any): any;
    };
```

**Comment by cristianbote** [comment]
@michalczaplinski thank you so much! @devrnt you're ok with going with @michalczaplinski's PR?

**Comment by devrnt** [comment]
Ofcourse! @cristianbote

**Comment by cristianbote** [comment]
published `goober@2.0.12` with the patch 👍
